### PR TITLE
Backporting to beta

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

**Comment by rphmeier** [comment]
This looks like it includes the changes in `next-hardfork` -- we shouldn't backport any of those until they're in.

File: ethcore/src/miner/banning_queue.rs [hunk 1]
```diff
@@ -0,0 +1,339 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
```
Comment (gavofyork) [inline ethcore/src/miner/banning_queue.rs]: Banning functionality shouldn't be making it into 1.3 series.

**Comment by gavofyork** [comment]
looks like the banning queue functionality was accidentally included here. it's not part of the EIPs and shouldn't be making it into 1.3 series.

File: ethcore/src/miner/banning_queue.rs (hunk 1 above)
Comment (arkpar) [inline ethcore/src/miner/banning_queue.rs]: Merg leftover. It is not even included in the build. Removed anyway

File: ethcore/build.rs [hunk 2]
```diff
@@ -18,6 +18,6 @@ extern crate ethcore_ipc_codegen;
 
 fn main() {
 	ethcore_ipc_codegen::derive_binary("src/types/mod.rs.in").unwrap();
-	ethcore_ipc_codegen::derive_ipc("src/client/traits.rs").unwrap();
```
Comment (gavofyork) [inline ethcore/build.rs]: don't remember changing this file...

File: ethcore/src/client/chain_notify.rs [hunk 3]
```diff
@@ -15,12 +15,10 @@
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
 use util::numbers::*;
-use ipc::{IpcConfig, BinaryConvertError};
```
Comment (gavofyork) [inline ethcore/src/client/chain_notify.rs]: don't remember changing this file

File: ethcore/src/client/mod.rs [hunk 4]
```diff
@@ -30,13 +30,17 @@ pub use self::test_client::{TestBlockChainClient, EachBlockWith};
 pub use types::trace_filter::Filter as TraceFilter;
 pub use executive::{Executed, Executive, TransactOptions};
 pub use env_info::{LastHashes, EnvInfo};
-pub use self::chain_notify::{ChainNotify, ChainNotifyClient};
```
Comment (gavofyork) [inline ethcore/src/client/mod.rs]: don't remember these changes.

File: ethcore/src/miner/transaction_queue.rs [hunk 5]
```diff
@@ -21,55 +21,6 @@
 //! transaction's nonce and next nonce expected from this sender). If nonces are equal transaction's gas price is used
 //! for comparison (higher gas price = higher priority).
 //!
-//! # Usage Example
```
Comment (gavofyork) [inline ethcore/src/miner/transaction_queue.rs]: don't remember removing all of this

**Comment by NikVolf** [comment]
@gavofyork you used `Option<u8>` in one of the new methods, and it cannot be serialized for ipc, so this ipc-ralted changes are backported from master where ipc codegen is not executed until feature is on (unlike it is in beta, where it is always executed)

**Comment by arkpar** [comment]
I've disabled IPC code generation in this PR because #2976 brakes it. IPC codegen is already disabled in master.

File: ipc/codegen/src/lib.rs [hunk 6]
```diff
@@ -49,7 +49,7 @@ include!(concat!(env!("OUT_DIR"), "/lib.rs"));
 include!("lib.rs.in");
 
 #[cfg(feature = "with-syntex")]
-pub fn register(reg: &mut syntex::Registry) {
+pub fn register_cleaner(reg: &mut syntex::Registry) {
```
Comment (gavofyork) [inline ipc/codegen/src/lib.rs]: don't remember changing this file.

File: ipc/hypervisor/src/service.rs.in [hunk 7]
```diff
@@ -34,7 +34,7 @@ pub struct HypervisorService {
 	check_list: RwLock<HashMap<IpcModuleId, bool>>,
 }
 
-#[derive(Ipc)]
+#[ipc]
```
Comment (gavofyork) [inline ipc/hypervisor/src/service.rs.in]: this and the similar changes following were not EIP related.

File: parity/main.rs [hunk 8]
```diff
@@ -119,11 +118,6 @@ fn start() -> Result<String, String> {
 fn main() {
 	// Always print backtrace on panic.
 	::std::env::set_var("RUST_BACKTRACE", "1");
-	// just redirect to the sync::main()
-	if std::env::args().nth(1).map_or(false, |arg| arg == "sync") {
```
Comment (gavofyork) [inline parity/main.rs]: not EIP related.

File: parity/modules.rs [hunk 9]
```diff
@@ -25,10 +25,6 @@ use self::ipc_deps::*;
 use ethcore_logger::Config as LogConfig;
 
 pub mod service_urls {
-	pub const CLIENT: &'static str = "ipc:///tmp/parity-chain.ipc";
```
Comment (gavofyork) [inline parity/modules.rs]: not EIP related

File: sync/src/api.rs [hunk 10]
```diff
@@ -40,7 +40,9 @@ pub struct SyncConfig {
 	/// Max blocks to download ahead
 	pub max_download_ahead_blocks: usize,
 	/// Network ID
-	pub network_id: U256,
+	pub network_id: usize,
+	/// Main "eth" subprotocol name.
+	pub subprotocol_name: [u8; 3],
```
Comment (gavofyork) [inline sync/src/api.rs]: don't think this additional line is related

File: sync/src/api.rs [hunk 11]
```diff
@@ -49,7 +51,8 @@ impl Default for SyncConfig {
 	fn default() -> SyncConfig {
 		SyncConfig {
 			max_download_ahead_blocks: 20000,
-			network_id: U256::from(1),
+			network_id: 1,
+			subprotocol_name: *b"eth",
```
Comment (gavofyork) [inline sync/src/api.rs]: `subprotocol_name` was from a different PR, i think

File: sync/src/api.rs [hunk 12]
```diff
@@ -90,8 +93,6 @@ impl EthSync {
 	}
 }
 
-#[derive(Ipc)]
-#[ipc(client_ident="SyncClient")]
```
Comment (gavofyork) [inline sync/src/api.rs]: not from the EIP PR

File: sync/src/api.rs [hunk 13]
```diff
@@ -185,8 +186,6 @@ pub trait ManageNetwork : Send + Sync {
 }
 
 
-#[derive(Ipc)]
```
Comment (gavofyork) [inline sync/src/api.rs]: not from the EIP PR

File: sync/src/lib.rs [hunk 14]
```diff
@@ -91,7 +91,7 @@ mod api {
 	include!(concat!(env!("OUT_DIR"), "/api.rs"));
 }
 
-pub use api::{EthSync, SyncProvider, SyncClient, NetworkManagerClient, ManageNetwork, SyncConfig,
+pub use api::{EthSync, SyncProvider, ManageNetwork, SyncConfig,
```
Comment (gavofyork) [inline sync/src/lib.rs]: not from the EIP PR

File: test.sh [hunk 15]
```diff
@@ -1,11 +1,11 @@
 #!/bin/sh
 # Running Parity Full Test Sute
 
-FEATURES="json-tests ipc"
+FEATURES="json-tests"
```
Comment (gavofyork) [inline test.sh]: not from the EIP PR

File: test.sh [hunk 16]
```diff
@@ -1,11 +1,11 @@
...
 
 case $1 in
     --no-json)
-    FEATURES="ipc"
+    FEATURES=""
```
Comment (gavofyork) [inline test.sh]: not from the EIP PR

**Comment by gavofyork** [comment]
quite a few changes from the EXP and some IPC changes seem to have been brought in - was that intentional?

**Comment by gavofyork** [comment]
seems the IPC changes were intentional - the partial EXP changes not so much?

**Comment by arkpar** [comment]
EXP support is already in beta. Looks like it was not properly merged before.

**Comment by gavofyork** [comment]
fair enough

**Pull request merged**